tests/gacha_tests.cpp

```diff
@@ -0,0 +1,162 @@
+
+   action_result winreward(account_name owner, uint64_t id, int64_t score, extended_asset value) {
+      return PUSH_ACTION(gacha_account_name, gacha_account_name, (owner)(id)(score)(value));
+   }
+
+   action_result raincheck(account_name owner, uint64_t id, int64_t score) {
+      return PUSH_ACTION(gacha_account_name, gacha_account_name, (owner)(id)(score));
+   }
+};
+
+BOOST_AUTO_TEST_SUITE(gxc_gacha_tests)
+
+BOOST_FIXTURE_TEST_CASE(open_tests, gxc_gacha_tester) try {
+
+   std::array<name, 2> schm = {N(gacha1), N(eun2ce)};
+   vector<grade> gd = {{.reward = asset::from_string("5000.00 ENC"), .score = 1, .limit = 0}};
+   string expiration = "2020-07-28T12:43:00";
+
+   subopen(schm, EA("5000.00 ENC@eun2ce"));
+   BOOST_REQUIRE_EQUAL(success(),
+         open(schm, gd, EA("5000.00 ENC@eun2ce"), expiration, 1)
+   );
+
+} FC_LOG_AND_RETHROW()
+
+BOOST_FIXTURE_TEST_CASE(issue_tests, gxc_gacha_tester) try {
+
+   vector<grade> gd = {{.reward = asset::from_string("5000.00 ENC"), .score = 1, .limit = 0}};
+   string expiration = "2018-01-01T00:00:10";
+   string dseedhash = "4855c552bbdd69cef5faee9bcc16a2f6a6b9d5f04560fa6a4045a5195c1f4f75";
+
+   BOOST_REQUIRE_EQUAL(wasm_assert_msg("scheme not found"),
+         issue(N(conr2d), {N(gacha123), N(eun2ce)}, dseedhash)
+   );
+
+   subopen({N(gacha2), N(eun2ce)}, EA("5000.00 ENC@eun2ce"));
+   open({N(gacha2), N(eun2ce)}, gd, EA("5000.00 ENC@eun2ce"), expiration, 1);
+   produce_blocks(1);
+
+   BOOST_REQUIRE_EQUAL(wasm_assert_msg("scheme expired"),
+         issue(N(conr2d), {N(gacha2), N(eun2ce)}, dseedhash)
+   );
+
+   expiration = "2022-01-01T00:00:10";
+   subopen({N(gacha3), N(conr2d)}, EA("5000.00 CRD@conr2d"));
+   open({N(gacha3), N(conr2d)}, gd, EA("5000.00 CRD@conr2d"), expiration, 1);
+   produce_blocks(1);
+
+   BOOST_REQUIRE_EQUAL(success(),
+         issue(N(eun2ce), {N(gacha3), N(conr2d)}, dseedhash)
+   );
+
+} FC_LOG_AND_RETHROW()
+
+BOOST_FIXTURE_TEST_CASE(setoseed_tests, gxc_gacha_tester) try {
+
+   vector<grade> gd = {{.reward = asset::from_string("5000.00 ENC"), .score = 1, .limit = 0}};
+   string expiration = "2022-01-01T00:00:10";
+   string dseedhash = "4855c552bbdd69cef5faee9bcc16a2f6a6b9d5f04560fa6a4045a5195c1f4f75";
+   string oseed = "4c519413ac98e5ead1c3b412e5d053ba0d57245e7689e6fcbe6dd9b81aa88dd7";
+
+   subopen({N(gachatest), N(eun2ce)}, EA("5000.00 ENC@eun2ce"));
+   open({N(gachatest), N(eun2ce)}, gd, EA("5000.00 ENC@eun2ce"), expiration, 1);
+   issue(N(conr2d), {N(gachatest), N(eun2ce)}, dseedhash, 1);
+   produce_blocks(1);
+
+   BOOST_REQUIRE_EQUAL(success(),
+         setoseed(1, oseed, N(conr2d))
+   );
+   produce_blocks(1);
+
+   string scmstr = "{\"scheme_name\": \"gachatest\", \"grades\": [{\"reward\": \"5000.00 ENC\", \"score\": 1, \"limit\": 0}], \"budget\": \"5000.00 ENC@eun2ce\", \"expiration\": \"2022-01-01T00:00:10\", \"precision\": 1, \"deadline_sec\": 604800, \"out\": \"0.00 ENC\", \"out_count\": [0], \"issued\": 1, \"unresolved\": 1}";
```

Comment:
You can use here raw string literal. You don't need to escape double quotation mark every time.

``` c++
string scmstr = R"({"scheme_name": "gachatest", ... })";
```

Or, have you tried passing grade struct to fc::variant directly?